New design for the maintenance mode

mnapoli opened the pull request:
Fixes #7620 

Pull request based on #7578, don't merge this one before merging #7578.

Current design:

![](https://cloud.githubusercontent.com/assets/720328/7059027/294b107a-debe-11e4-9b76-81119838ead2.png)

There's not much to say in that page so it feels a bit empty…

mnapoli committed:
#7450 First step at redesigning the updater

mnapoli committed:
#7450 Redesigning the updater

mnapoli committed:
Redesign the error page as it uses the same style as the updater

mnapoli committed:
#7450 Redesigning the database updater

mnapoli committed:
#7450 Redesigning the database updater success screen

mnapoli committed:
ref #7450 Redesigning the database updater error

mnapoli committed:
#7450 Make the donation box look better during the update

mnapoli committed:
#7620 Redesign of the maintenance mode

mattab commented:
the visual looks good to me, a solid improvement!
 
(I didn't look at code)

mnapoli commented:
FYI the diff is huge because this PR is based on the other PR which is also based on another PR :)

But the commit is very simple: 71c7a7d4d2f29634c621d5aaedaa8eee09880a01

tsteur commented on plugins/CoreUpdater/templates/layout.twig:
```diff
@@ -42,13 +42,17 @@
-    </div>
+
+<div class="logo">
+    <img title="Piwik" alt="Piwik" src="plugins/Morpheus/images/logo-header.png"/>
```
Should we show a possible custom logo here?

mnapoli replied:
This is the diff for #7578 (updater redesign).

I've added support for custom logo in the maintenance mode, I'll try to add support for that in the updater in #7578.

mnapoli replied:
Added support for custom logos in the updater too (in #7578).

mnapoli committed:
#7620 Take into account custom logos for the maintenance mode

mnapoli committed:
Add support for custom logo in the updater

mattab commented on core/FrontController.php:
```diff
@@ -412,28 +413,27 @@ protected function prepareDispatch($module, $action, $parameters)
+            $logo = new CustomLogo();
+            $logoUrl = $logo->getHeaderLogoUrl();
+            $faviconUrl = $logo->getPathUserFavicon();
+        } catch (Exception $ex) {
```
+1 - I tested that maintenance is displayed correctly when DB is not accessible and it works thanks to this catch (case where the DB could be down during maintenance)

mnapoli commented:
Keep in mind while reviewing that this PR is based on #7578, only the 2 last commits are about the maintenance.

mattab committed:
Although Piwik is always free as in freedom (http://piwik.org/free-software/),
it may not be free as in beer - by removing those titles we remove some possible confusion
 (eg. we don't show "Free Web Analytics" on the login form on Piwik Cloud)

mattab commented:
* the subtitle we can use Piwik::getRandomTitle() instead of `free/libre analytics platform`
 * otherwise looks good +1 to merge

mnapoli committed:
In the maintenance mode use the Piwik titles

mnapoli commented:
It now uses `Piwik::getRandomTitle()`

mnapoli commented:
@mattab I'm afraid the commits are not good here :/ I was rebasing this PR because as you can see the diff includes some commits of the updater redesign, which is not good. I don't know if there are any consequences
